src/java.base/share/classes/java/nio/channels/FileChannel.java

```diff
@@ -588,6 +588,10 @@ public final long write(ByteBuffer[] srcs) throws IOException {
      * operating systems can transfer bytes directly from the filesystem cache
      * to the target channel without actually copying them.  </p>
      *
+     * @apiNote
+     * This method only returns the number of bytes that were actually
+     * transferred, it will not return -1.
+     *
```

Comment:
The transferTo method is file -> target channel and I think it would be confusing to mention -1 here.
We could add an apiNote to point out that the method returns zero if the given position is greater than the file's current size, and also zero if the target channel is non-blocking and has no bytes free in its output buffer, but I think the javadoc is already clear.

Reply:
OK, I will remove this @apiNote addition